gadgets/trace_tcp/program.bpf.c

```diff
@@ -0,0 +1,406 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 Microsoft Corporation
+//
+// Based on tcptracer(8) from BCC by Kinvolk GmbH and
+// tcpconnect(8) by Anton Protopopov
+#include <vmlinux.h>
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+#include <gadget/mntns_filter.h>
+#include <gadget/types.h>
+#include <gadget/macros.h>
+#include <gadget/buffer.h>
+
+/* The maximum number of items in maps */
+#define MAX_ENTRIES 8192
+
+#define TASK_COMM_LEN 16
+
+enum event_type : u8 {
+	TCP_EVENT_TYPE_CONNECT,
+	TCP_EVENT_TYPE_ACCEPT,
+	TCP_EVENT_TYPE_CLOSE,
+};
```

Comment:
Remove the `TCP_EVENT_TYPE_` prefix to make it prettier on the output? Otherwise the "real data" can be trimmed if the screen is not big enough:

```bash 
[sudo] password for mvb:           
INFO[0000] Experimental features enabled                
RUNTIME.CONTAINERNAME     SRC                      DST                      TASK          TS_US         PID           UID           GID           TYPE        
myfoo2                    172.17.0.3:645           142.250.217.206:20480    wget          19263843177   71597         0             0             TCP_EVENT_T…
myfoo2                    172.17.0.3:645           142.250.217.206:20480    wget          19264094134   71597         0             0             TCP_EVENT_T…
myfoo2                    172.17.0.3:61657         142.250.217.196:20480    wget          19264224145   71597         0             0             TCP_EVENT_T…
myfoo2                    172.17.0.3:61657         142.250.217.196:20480    wget          19264617302   71597         0             0             TCP_EVENT_T…
```

Another option would be to use "trim left" on the matadata, but IMO this prefix doesn't add much value on this case. (It's useful on bigger applications to avoid a collision with other identifier, but in this small program that shouldn't be an issue)

Reply:
✅